### ISSUE TITLE
Loss becomes NaN during training reproduction

### ISSUE THREAD

**Issue by lonelyqian** [NikoNairre/CT-CIG#4]
<img width="1629" height="816" alt="Image" src="https://github.com/user-attachments/assets/feae6aeb-b491-485d-abb4-8e2fbc2963b4" />
When reproducing the training process, the training loss (loss_avr) becomes NaN at multiple steps throughout the training. The loss curve shows several sharp spikes and drops, and in some cases, the loss value is completely invalid (NaN), which halts the training process.
I wonder if you or your team have ever encountered similar unstable loss issues during the training process? If it's convenient, could you share the original training logs (including loss curves, learning rate settings, and any key training parameters) for reference? This would be extremely helpful for me to locate the root cause of the NaN loss problem.

**Comment by NikoNairre** [comment]
Indeed, this loss contains NAN values, apologize for this fault. I'm also wondering about the root of the problem, and I'd like to find solutions to tackle this problem if possible.

**Comment by NikoNairre** [comment]
A possible solution might be altering the precision of parameters from fp16 to fp32 in backward propagation.
